rebecca-millerr/cs21-platformer
# basic platforms

# Basic platform setup

## Finished

- HTML canvas representation of map
- Canvas with adjustable dimensions and scroll speed
- Map infinitely scrolls to the left (as runners run to the right)
- For MVP, always have a default map with just a floor and nothing else

## Still TODO
- Map magically will receive updates _from_ server every frame for single source of truth
- Make canvas clickable by builders, and then...
- ...map magically will send updates _to_ server every time a builder adds a block
- Add physics to blocks

## Lingering thoughts

Right now the new column gets added on the front end, i.e. individually for each client (which is the only way to get this PR to work with no Erlang server). Even though every user does get the same right column every time, I'm thinking this could cause issues with this updating the server a bunch of times (once for each client), so maybe in the future we want this to actually happen on the server and then update each client... 

## To test

Run the app, then go to /game. Voila! It doesn't look like much right now because it just creates the same map infinitely, but see video below for more obvious example of this in action. I changed it temporarily so the new columns come in with red blocks instead of blue, which makes it clearer that new columns are actually being added as older ones are deleted. 

https://user-images.githubusercontent.com/52360419/162635315-c021844e-4024-4b74-9285-ad948e9cb664.mov

Super appreciate any thoughts <3
